common/scala/src/main/scala/org/apache/openwhisk/core/connector/Message.scala

```diff
@@ -340,7 +345,22 @@ object Activation extends DefaultJsonProtocol {
       "conductor",
       "memory",
       "causedBy",
-      "size")
+      "size",
+      "actionStatusCode")
+
+  /** Get "StatusCode" from result response **/
+  def getActivationStatusCode(result: Option[JsValue]): Option[Int] = {
+    val statusCode = JsHelpers.getFieldPath(result.get.asJsObject, ERROR_FIELD, "statusCode")
+
+    statusCode match {
+      case Some(value) => Some(value.convertTo[Int])
+      case None =>
+        JsHelpers.getFieldPath(result.get.asJsObject, "statusCode") match {
+          case Some(value) => Some(value.convertTo[Int])
+          case None        => None
+        }
```

Comment:
```
statusCode
.map(_.intValue)
.orElse(_ => JsHelpers.getFieldPath(result.get.asJsObject, "statusCode").map(_.intValue))
```

i think - warning i didn't vet this in an ide/compiler but see my comment above, i don't think either of these are sufficient.

Reply:
Sure, will try your suggestion.